### PR TITLE
docs(contribution-guidelines): add github issue and pull request guidelines and templates

### PR DESCRIPTION
**Pull request by JeremyFriesenGitHub**
# Description

Describe the PR with the following guidelines

## What changes does this PR add?

This PR adds PR guidelines and a PR checklist to the writing documentation page on the docs site. Additionally, PR templates have also been created as well (in the .github directory).  

## **Why do we need to add to the docs? How does this PR address the issue(s)?**

This tries to attempt to make decent writing guidelines and solve the todo's for that page specifically. Automated PR templates are also a must and can help create contribution consistency among developpers.  

## Dependency Changes (if applicable)

N/A

## Before and After Screenshots (if applicable)
Here is a before and after for the writing documentation page:

|        Before         |        After         |
| :-------------------: | :------------------: |
| image | image |

## Additional comments (if applicable)

I've put this in draft because I wanted input on this so far (in terms of frontend layout, resources that we can use, etc.) I know right now this is not very pretty (nor perhaps as informative as it should be). Its definitely an issue I think that takes a strong second opinion for review. This is linked to issue #77. Issue #78 is also very likely to be incorporated into this PR as well, since its similar and also a todo for the writing documentation.

## Summary by Sourcery

Enhance the contribution guidelines by adding comprehensive pull request guidelines, issue guidelines, and templates for features, tests, and documentation. Update the documentation to include a checklist for code quality and PR review, and remove the outdated pull request template.

Documentation:
- Add detailed guidelines for creating pull requests, including a checklist for code quality and PR review, to the contribution guidelines documentation.
- Introduce a section on finding the correct type for commits, issues, and pull requests, with definitions for various types like fix, feat, docs, test, chore, style, refactor, perf, build, ci, and revert.
- Provide guidelines for writing GitHub issues, including title, description, labels, and assignees.

Chores:
- Remove the old pull request template file from the .github directory.

## Summary by CodeRabbit

- **New Features**
	- Introduced a "Pull Request Guidelines" section detailing best practices for submitting PRs.
	- Added a "Pull Request Checklist" to assist contributors in ensuring code quality before submission.

- **Documentation**
	- Enhanced documentation guidelines with a focus on categorizing contributions and best practices for writing GitHub issues.
	- Removed the "index" page from the navigation of the Contribution Guidelines and Knowledge Base sections to improve clarity and user navigation.
	- Simplified the Tools Overview documentation structure by removing the "pages" key from the metadata.

### CODE REVIEW

**Comment by JeremyFriesenGitHub** [comment]
i think for the most part (issues + prs + commits including), ppl don't know the correct type to use (difference between a fix vs a chore), so I think clearly explaining/defining them in a page would be very beneficial. A chart decision chart could also be helpful with this as well..

**Comment by JeremyFriesenGitHub** [comment]
also perhaps the frontend pr stuff should go with the pull requests section

**Comment by JeremyFriesenGitHub** [comment]
> also I'm thinking that the PR guidelines should go into the Pull Requests page, instead of having it in writing documentation

I just went ahead and did this

**Comment by JeremyFriesenGitHub** [comment]
I'm also wondering what is the convention when naming issues (perhaps even prs as well), the templates currently setup ``[Type]`` but I don't know if this should be the standard or not (considering that some have also been named similar to their PRs).

**Comment by JeremyFriesenGitHub** [comment]
> I'm also wondering what is the convention when naming issues (perhaps even prs as well), the templates currently setup `[Type]` but I don't know if this should be the standard or not (considering that some have also been named similar to their PRs).

this would also help with maybe creating more issue templates as well

**Comment by MFarabi619** [comment]
> I'm also wondering what is the convention when naming issues (perhaps even prs as well), the templates currently setup `[Type]` but I don't know if this should be the standard or not (considering that some have also been named similar to their PRs).

What I was thinking is to keep it simple like:

### Issue:
`feat(ui/portal): add sign in button to navbar`

### PR:
`feat(ui/portal): add sign in button to navbar`

### Branch:
`jfriesen/feat-ui-portal/84-add-sign-in-button-to-navbar`

or 

`jfriesen/feat-ui-portal/84-add-sign-in-button-to-navbar`

Let me know what you think

File: .github/PULL_REQUEST_TEMPLATE/docs.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+---
```
Comment (MFarabi619) [inline .github/PULL_REQUEST_TEMPLATE/docs.md:1]: Nice

File: .github/PULL_REQUEST_TEMPLATE/feature.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+---
```
Comment (MFarabi619) [inline .github/PULL_REQUEST_TEMPLATE/feature.md:1]: Nice

File: .github/PULL_REQUEST_TEMPLATE/test.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
+---
```
Comment (MFarabi619) [inline .github/PULL_REQUEST_TEMPLATE/test.md:1]: Nice

File: apps/docs/src/content/docs/contribution-guidelines/pull-requests.mdx [hunk 4]
```diff
@@ -8,10 +8,79 @@ description: The diplomatic way to sneak your changes into main.
...
 
-Ensure that the issue is linked to your PR, and is part of a milestone. However, do not add it to the project board, because it shows up as a card and causes visual pollution. We can easily just see your linked PR from the closed issue if we need to find it.
+## Pull Request Guidelines
+
+When creating a Pull Request (PR), be mindful about what you're trying to contribute and the type of contribution: 
+- Be sure to checkout [Branching Guidelines](/docs/contribution-guidelines/branching-guidelines), [Coding Guidelines](/docs/contribution-guidelines/coding-guidelines) and [Commit Conventions](/docs/contribution-guidelines/conventional-commits) before making a PR.
+- Use the appropriate template that matches with your branch type.
+- Ensure your PR scope and subject align with your main commit(s).
+- If your code isn’t fully ready to be merged, consider creating a draft PR instead.
```
Comment (MFarabi619) [inline apps/docs/src/content/docs/contribution-guidelines/pull-requests.mdx:23]: Very nice

**Comment by JeremyFriesenGitHub** [comment]
> > I'm also wondering what is the convention when naming issues (perhaps even prs as well), the templates currently setup `[Type]` but I don't know if this should be the standard or not (considering that some have also been named similar to their PRs).
> 
> 
> 
> What I was thinking is to keep it simple like:
> 
> 
> 
> ### Issue:
> 
> `feat(ui/portal): add sign in button to navbar`
> 
> 
> 
> ### PR:
> 
> `feat(ui/portal): add sign in button to navbar`
> 
> 
> 
> ### Branch:
> 
> `jfriesen/feat-ui-portal/84-add-sign-in-button-to-navbar`
> 
> 
> 
> or 
> 
> 
> 
> `jfriesen/feat/ui/portal/84-add-sign-in-button-to-navbar`
> 
> 
> 
> Let me know what you think

Ok cool, it was just issues I was wondering about. Also, I didn't know that scopes could be included in branches. Would that be recommended, or is just listing name/type/issue-number-and-name ok?

**Comment by MFarabi619** [comment]
@sourcery-ai review

File: apps/docs/src/content/docs/contribution-guidelines/pull-requests.mdx [hunk 5]
```diff
@@ -8,10 +8,79 @@ description: The diplomatic way to sneak your changes into main.
...
 
 Do your best to close a single issue per pull request, and always use the exact same PR title as the issue name. This makes debugging easier later. If the current issue is blocking the next, simply create a new branch off the branch for the pull request and continue working on the next issue on that branch. This technique is called [branch stacking](https://medium.com/@lneves12/git-how-to-stack-multiple-git-branches-and-rebase-them-like-a-pro-91c0cdf67ef). With the power of `git rebase`, you can continually apply PR feedback for the first branch while making progress for the next task on the second branch. Once again, [Lazygit](https://github.com/jesseduffield/lazygit) makes this very quick and easy to do cleanly.
 
-Ensure that the issue is linked to your PR, and is part of a milestone. However, do not add it to the project board, because it shows up as a card and causes visual pollution. We can easily just see your linked PR from the closed issue if we need to find it.
+## Pull Request Guidelines
```
[the remark this answers is not in the document]
Comment (JeremyFriesenGitHub) [reply apps/docs/src/content/docs/contribution-guidelines/pull-requests.mdx:17]: so i commented out the checklist and then removed templates. They are now in .github/PULL_REQUEST_TEMPLATE/ dir.

[the remark this answers is not in the document]
Comment (JeremyFriesenGitHub) [reply apps/docs/src/content/docs/contribution-guidelines/pull-requests.mdx:17]: yes, i did this as well. I have created different templates for features, tests and docs.

[the remark this answers is not in the document]
Comment (JeremyFriesenGitHub) [reply apps/docs/src/content/docs/contribution-guidelines/pull-requests.mdx:17]: i was wondering if i should keep the checklist in this page or should just remove it altogether

[the remark this answers is not in the document]
Comment (JeremyFriesenGitHub) [reply apps/docs/src/content/docs/contribution-guidelines/pull-requests.mdx:17]: so ill keep it int the docs page,, and then remove it from the templates for now

[the remark this answers is not in the document]
Comment (JeremyFriesenGitHub) [reply apps/docs/src/content/docs/contribution-guidelines/pull-requests.mdx:17]: ok, lastly you haven't checked my newest commits...

[the remark this answers is not in the document]
Comment (JeremyFriesenGitHub) [reply apps/docs/src/content/docs/contribution-guidelines/pull-requests.mdx:17]: can you review my latest commits? especially the ones for github issues

**Comment by JeremyFriesenGitHub** [comment]
todo: fix/create issues

**Comment by MFarabi619** [comment]
> > > I'm also wondering what is the convention when naming issues (perhaps even prs as well), the templates currently setup `[Type]` but I don't know if this should be the standard or not (considering that some have also been named similar to their PRs).
> > 
> > 
> > What I was thinking is to keep it simple like:
> > ### Issue:
> > `feat(ui/portal): add sign in button to navbar`
> > ### PR:
> > `feat(ui/portal): add sign in button to navbar`
> > ### Branch:
> > `jfriesen/feat-ui-portal/84-add-sign-in-button-to-navbar`
> > or
> > `jfriesen/feat/ui/portal/84-add-sign-in-button-to-navbar`
> > Let me know what you think
> 
> Ok cool, it was just issues I was wondering about. Also, I didn't know that scopes could be included in branches. Would that be recommended, or is just listing name/type/issue-number-and-name ok?

Yes scopes could be included in branches. I'm not sure how `recommended` it would be, I haven't seen many projects do it. It does give us more specificity into what the purpose of the branch is though; the challenge is that if the issue name changes we now need to propagate those changes to the PR and the branch as well. 

What do you think?

**Review by MFarabi619 — DISMISSED** [review]
Great PR, docs are very nicely written and formatted, and thanks for adding the templates too :pray:

File: .github/PULL_REQUEST_TEMPLATE/continuous-integration--ci.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+If there are any comments to add, list them here.
+
+# ✅ Checklist
+
+Make sure your PR follows the according checklist:
+
+### **Code Quality**
+
+<table>
```
Comment (HasithDeAlwis) [inline .github/PULL_REQUEST_TEMPLATE/continuous-integration--ci.md:41]: Very small nitpick, I don't really mind, but would like your opinion on this as well. We can use MD to write these:
Example:
- [ ] checkbox

I did this with "- [ ] checkbox"
Lmk your thoughts

File: .github/PULL_REQUEST_TEMPLATE/docs.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+name: Docs
+about: Add to the docs for this project
+title: "docs(scope-of-PR): description of PR"
+labels: 'documentation'
+assignees: ''
+---
+
+# 🚩 Linked issue
+
+**Please remember to link the according issue here in this section with a [hyperlink](https://github.com/cuhacking/2025/issues).** This will give reviewers a chance to look to see what problem you are trying to solve.
```
Comment (HasithDeAlwis) [inline .github/PULL_REQUEST_TEMPLATE/docs.md:11]: nice touch 🔥

File: apps/docs/src/content/docs/contribution-guidelines/pull-requests.mdx [hunk 8]
```diff
@@ -8,10 +8,109 @@ description: The diplomatic way to sneak your changes into main.
...
 
-Ensure that the issue is linked to your PR, and is part of a milestone. However, do not add it to the project board, because it shows up as a card and causes visual pollution. We can easily just see your linked PR from the closed issue if we need to find it.
+Do your best to close a single issue per pull request, and always use the same PR title as the issue name. This makes debugging easier later. If the current issue is blocking the next, simply create a new branch off the branch for the pull request and continue working on the next issue on that branch. This technique is called [branch stacking](https://medium.com/@lneves12/git-how-to-stack-multiple-git-branches-and-rebase-them-like-a-pro-91c0cdf67ef). With the power of `git rebase`, you can continually apply PR feedback for the first branch while making progress for the next task on the second branch. Once again, [Lazygit](https://github.com/jesseduffield/lazygit) makes this quick and easy to do cleanly.
```
Comment (HasithDeAlwis) [inline apps/docs/src/content/docs/contribution-guidelines/pull-requests.mdx:15]: nice link to the medium article 👍

**Review by HasithDeAlwis — APPROVED** [review]
Very, very nice work on this 👍

File: .github/PULL_REQUEST_TEMPLATE/continuous-integration--ci.md (hunk 6 above)
Comment (JeremyFriesenGitHub) [reply .github/PULL_REQUEST_TEMPLATE/continuous-integration--ci.md:41]: does not work in mdx...

Comment (JeremyFriesenGitHub) [reply .github/PULL_REQUEST_TEMPLATE/continuous-integration--ci.md:41]: or at least you won't be able to interact with the checkbox at all

Comment (HasithDeAlwis) [reply .github/PULL_REQUEST_TEMPLATE/continuous-integration--ci.md:41]: Really, on my end I can interact with it?

**Comment by JeremyFriesenGitHub** [comment]
> Very, very nice work on this 👍

thx, still tons left actually on the docs, but we need to at least get this through right now for better issue + pr documentation

File: .github/PULL_REQUEST_TEMPLATE/continuous-integration--ci.md (hunk 6 above)
Comment (JeremyFriesenGitHub) [reply .github/PULL_REQUEST_TEMPLATE/continuous-integration--ci.md:41]: weird, for me it was not working nicely at all...

Comment (HasithDeAlwis) [reply .github/PULL_REQUEST_TEMPLATE/continuous-integration--ci.md:41]: How does it work in this comment?

Comment (JeremyFriesenGitHub) [reply .github/PULL_REQUEST_TEMPLATE/continuous-integration--ci.md:41]: yes, but mdx is not rly md, i talked to farabi abt it and he said it was normal

Comment (HasithDeAlwis) [reply .github/PULL_REQUEST_TEMPLATE/continuous-integration--ci.md:41]: Yea, i'm aware, but from waht I know checkboxes work in MDX and MD. 

Refer to the PR below and you can interact with the checkboxes: 
PR #60 is a good example

**Review by HasithDeAlwis — APPROVED** [review]

**Review by MFarabi619 — APPROVED** [review]

**Pull request merged**